tests/components/deako/test_light.py

```diff
@@ -0,0 +1,275 @@
+"""Tests for the light module."""
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from homeassistant.components.deako.const import DOMAIN
+from homeassistant.components.deako.light import async_setup_entry
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    DOMAIN as LIGHT_DOMAIN,
+    ColorMode,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from .conftest import MockDeakoDevices
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.usefixtures("init_integration")
+@pytest.mark.asyncio
+async def test_light_setup_no_devices(
+    hass: HomeAssistant,
+    pydeako_deako_mock: MagicMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test light platform setup with no devices returned."""
+
+    # no devices
+    pydeako_deako_mock.get_devices.return_value = {}
+
+    light_entities = []
+
+    async def mock_add_entities(entities):
+        for entity in entities:
+            light_entities.append(entity)
+
+    await async_setup_entry(hass, mock_config_entry, mock_add_entities)
+
+    assert len(light_entities) == 0  # none have been added
+    pydeako_deako_mock.disconnect.assert_called_once()
+
+
+@pytest.mark.usefixtures("init_integration")
+@pytest.mark.asyncio
+async def test_light_setup_with_devices(
+    hass: HomeAssistant,
+    pydeako_deako_mock: MagicMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test light platform setup with devices returned."""
+
+    pydeako_deako_mock.get_devices.return_value = {
+        "some_device": {},
+    }
+
+    light_entities = []
+
+    def mock_add_entities(entities):
+        for entity in entities:
+            light_entities.append(entity)
+
+    await async_setup_entry(hass, mock_config_entry, mock_add_entities)
+
+    assert len(light_entities) == 1
+    pydeako_deako_mock.set_state_callback.assert_called_once_with(
+        "some_device", light_entities[0].on_update
+    )
+
+
+@pytest.mark.parametrize(("device_name", "device_uuid"), MockDeakoDevices().get_names())
+@pytest.mark.usefixtures("mock_deako_devices", "init_integration")
+@pytest.mark.asyncio
+async def test_light_initial_props(
+    hass: HomeAssistant,
+    pydeako_deako_mock: MagicMock,
+    entity_registry: er.EntityRegistry,
+    device_registry: dr.DeviceRegistry,
+    mock_devices: MockDeakoDevices,
+    device_name: str,
+    device_uuid: str,
+) -> None:
+    """Test lights are setup with accurate initial properties."""
+    # make sure our initial state exists
+    assert (state := hass.states.get(f"{LIGHT_DOMAIN}.{device_name}"))
+    assert (
+        state.state == STATE_ON
+        if mock_devices.get_state(device_uuid)["power"]
+        else STATE_OFF
+    )
```

Comment:
Please don't have conditions in the tests. This makes it harder what we are testing.